index.d.ts

```diff
@@ -74,3 +68,21 @@ export interface Account {
   // The provider may have captured a display name
   displayName?: string
 }
+
+export interface PolkadotSigner {
+  publicKey: Uint8Array
+  sign: (
+    callData: Uint8Array,
+    signedExtensions: Record<
+      string,
+      {
+        identifier: string
+        value: Uint8Array
+        additionalSigned: Uint8Array
+      }
+    >,
+    metadata: Uint8Array,
+    atBlockNumber: number,
+    hasher?: (data: Uint8Array) => Uint8Array,
+  ) => Promise<Uint8Array>
+}
```

Comment:
This type might be an implementation detail for Polkadot Provider implementations.
For example, if `Account` is
```ts
export interface Account {
  address: string
  publicKey: Uint8Array
  displayName?: string
  sign(callData: Uint8Array, hintedSignedExtensions: TypeTBD, hasher?: (data: Uint8Array) => Uint8Array,)
}

export interface Chain {
  getAccounts(): Promise<Account[]>
}

```

Then, the consumer of the Polkadot Provider API, will do
```ts
const chain = await provider.getChain(chainId)
const accounts = await chain.getAccounts()
const account = accounts[0]
const signedExtrinsic = account.sign(callData, hintedSignedExtension);
```
The `Account.sign()` method will have enough context to create a `PolkadotSigner` because it knows the chain it belongs to fetch metadata and available signed extensions.

WDYT @ryanleecode?

Reply:
Oops yeah you're right. I think this is the right one

```ts
interface PolkadotSigner {
    publicKey: Uint8Array;
    sign: (callData: Uint8Array, signedExtensions: Record<string, {
        identifier: string;
        value: Uint8Array;
        additionalSigned: Uint8Array;
    }>, metadata: Uint8Array, atBlockNumber: number, hasher?: (data: Uint8Array) => Uint8Array) => Promise<Uint8Array>;
}
```